docs/sources/operators-guide/deploying-grafana-mimir/jsonnet/memberlist.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "Migrating from Consul to Memberlist KV store for hash rings"
+menuTitle: "Migrating from Consul to Memberlist"
+description: "Learn how to migrate from using Consul as KV store for hash rings to using memberlist without any downtime."
+weight: 30
+---
+
+# Migrating from Consul to Memberlist as KV store for hash rings without downtime
+
+Mimir Jsonnet uses `memberlist` as KV store for hash rings, unless disabled by using
+
+```jsonnet
+{
+  _config+:: {
+    memberlist_ring_enabled: false
+  }
+}
+```
```

Comment:
would it make sense to mention somewhere in this document which version of jsonnet that this is guaranteed to work with?

I'm asking that because I'm guessing that at some point in the future we might make changes to jsonnet which would break this migration as it is documented, so how can we ensure that in the future users won't follow this guide and end up having a bad experience because we changed jsonnet in the meantime?

Reply:
My understanding is that this will be publised on https://grafana.com/docs/mimir/ website, together with version number. If this procedure changes in the future, we will update it.

> Do we rely on persons who make changes to the mimir jsonnet in the future thinking about this document and updating it accordingly?

Relevant changes would need to happen in `memberlist.libsonnet`. Yes, I think person updating it should change this article too.